CMakeLists.txt

```
@@ -29,6 +28,13 @@ option(SLEEF_DISABLE_SSL "Disable testing with the SSL library" OFF)
 option(SLEEF_ENABLE_CUDA "Enable CUDA" OFF)
 option(SLEEF_ENABLE_CXX "Enable C++" OFF)
 
+set(_INTERNAL_SLEEF_BUILD_SHARED_LIBS)
+if(NOT DEFINED ${SLEEF_BUILD_SHARED_LIBS})
+  set(_INTERNAL_SLEEF_BUILD_SHARED_LIBS ${BUILD_SHARED_LIBS})
+else()
+  set(_INTERNAL_SLEEF_BUILD_SHARED_LIBS ${SLEEF_BUILD_SHARED_LIBS})
+endif()
+
```

Can this be done with options instead? For instance, 
```
option(BUILD_SHARED_LIBS "Build sleef as shared library" ON)
option(SLEEF_BUILD_SHARED_LIBS "Build sleef as shared library" ${BUILD_SHARED_LIBS})
```

---

@blapie -- These are not equivalent because this defines `SLEEF_BUILD_SHARED_LIBS` as a cache variable when no variable by that name (normal or cache) exists. So then changing `BUILD_SHARED_LIBS` only works as expected on the first configuration.

My blog post suggests the following:

```suggestion
if (DEFINED SLEEF_BUILD_SHARED_LIBS)
  set(BUILD_SHARED_LIBS ${SLEEF_BUILD_SHARED_LIBS})
endif ()
```

Then you can just look at the truthiness of `BUILD_SHARED_LIBS` down the line. If you care about documenting this variable in the CMake CLI/GUI, then this approach works:

```suggestion
set(SLEEF_BUILD_SHARED_LIBS "undefined"
    CACHE BOOL "Override BUILD_SHARED_LIBS while configuring SLEEF")
if (NOT SLEEF_BUILD_SHARED_LIBS STREQUAL "undefined")
  set(BUILD_SHARED_LIBS ${SLEEF_BUILD_SHARED_LIBS})
endif ()